File: graphql/graphql-quarkus-with-panache/src/main/java/org/example/graphql/model/Person.java
package org.example.graphql.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import org.hibernate.validator.constraints.Length;

import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;

@Entity
public class Person extends PanacheEntity {

    @Length(min = 8)
    private String name;

    @Enumerated(value = EnumType.STRING)
    private Gender gender;

    public Person() {
    }

    public Person(String name, Gender gender) {
        this.name = name;
        this.gender = gender;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Gender getGender() {
        return gender;
    }

    public void setGender(Gender gender) {
        this.gender = gender;
    }
}

File: quarkus/quarkus-hibernate-panache-kotlin/src/main/kotlin/org/example/model/Person.kt
package org.example.model

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Table
class Person : PanacheEntity() {

    companion object : PanacheCompanion<Person, Long> {}

    @ManyToOne
    lateinit var gender: Gender

}